demo-app/src/lib/trino-uqi-client.ts

```diff
@@ -43,6 +43,20 @@ async function createTrinoUqiClient(config: TrinoConfig): Promise<UqiClient> {
   ): Promise<AsyncIterableIterator<UqiResult>> {
     const queryIterator: TrinoIterator<QueryResult> =
       await context.client.query(q)
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const queryId = queryIterator['iter'].queryResult.id
+    let resultsProcessed = 0
+    let totalResults = 1
+
+    async function updateStatus(): Promise<void> {
+      const status = (await context.client.queryInfo(queryId)) as any
+      if (totalResults === 1) {
+        totalResults = (await status.queryStats.outputPositions) || 0
+      }
```

Comment:
Oooh, this looks cool. I don't remember this from our :pear:, can you tell me more about outputPositions? Sorry if my memory is lossy around this 😂

Reply:
To be honest, I think my answer is going to be more confusing than clarifying.  We just found it by console.logging a bunch of stuff and seeing what was there -- I can't even figure out where it's coming from.

When you and I paired on implementing this in our other app, we also used `outputPositions`, but got it from a slightly different place - `status['outputStage'].stageStats.outputPositions` vs. here it's `status.queryStats.outputPositions`.  If I'm remembering right, the only reason I looked in a different place was because the former wasn't working, but I think what actually fixed it was adding the `as any` as the type for the status -- so I'm not sure of the different between the two different `outputPositions`, but the first one may have also been fine to use.

Not sure if it's helpful, but here's what an example of `status` looks like when running the Trino demo query in Open-Truss (of course it would be great to be able to use `status.progressPercentage`, but we weren't getting `queryInfo` in time for that to be accurate...in fact I'm actually surprised to see that at anything other than 100%)

```
{
  status: {
    queryId: '20240711_192310_13170_mtmqs',
    session: {
      [not relevant, snip]
    },
    state: 'FINISHED',
    self: 'http://[REDACTED]/v1/query/20240711_192310_13170_mtmqs',
    fieldNames: [ 'name', 'phone', 'acctbal', 'comment' ],
    query: 'SELECT name, phone, acctbal, comment FROM tpch.sf1.customer limit 10\n',
    queryStats: {
      createTime: '2024-07-11T19:23:10.752Z',
      executionStartTime: '2024-07-11T19:23:10.753Z',
      lastHeartbeat: '2024-07-11T19:23:10.795Z',
      endTime: '2024-07-11T19:23:10.795Z',
      elapsedTime: '43.12ms',
      queuedTime: '172.00us',
      resourceWaitingTime: '1.32ms',
      dispatchingTime: '25.07us',
      executionTime: '41.60ms',
      analysisTime: '1.32ms',
      planningTime: '4.33ms',
      planningCpuTime: '3.58ms',
      finishingTime: '196.19us',
      totalTasks: 21,
      runningTasks: 4,
      completedTasks: 17,
      failedTasks: 0,
      totalDrivers: 85,
      queuedDrivers: 41,
      runningDrivers: 3,
      blockedDrivers: 0,
      completedDrivers: 41,
      cumulativeUserMemory: 30681052.022062004,
      failedCumulativeUserMemory: 0,
      userMemoryReservation: '246456B',
      revocableMemoryReservation: '0B',
      totalMemoryReservation: '246456B',
      peakUserMemoryReservation: '1649917B',
      peakRevocableMemoryReservation: '0B',
      peakTotalMemoryReservation: '1649917B',
      peakTaskUserMemory: '529525B',
      peakTaskRevocableMemory: '0B',
      peakTaskTotalMemory: '529525B',
      scheduled: true,
      progressPercentage: 48.23529411764706,
      runningPercentage: 3.5294117647058822,
      totalScheduledTime: '372.00ms',
      failedScheduledTime: '0.00ns',
      totalCpuTime: '362.00ms',
      failedCpuTime: '0.00ns',
      totalBlockedTime: '80.00ms',
      fullyBlocked: true,
      blockedReasons: [],
      physicalInputDataSize: '0B',
      failedPhysicalInputDataSize: '0B',
      physicalInputPositions: 60000,
      failedPhysicalInputPositions: 0,
      physicalInputReadTime: '0.00ns',
      failedPhysicalInputReadTime: '0.00ns',
      internalNetworkInputDataSize: '1825B',
      failedInternalNetworkInputDataSize: '0B',
      internalNetworkInputPositions: 20,
      failedInternalNetworkInputPositions: 0,
      rawInputDataSize: '0B',
      failedRawInputDataSize: '0B',
      rawInputPositions: 60000,
      failedRawInputPositions: 0,
      processedInputDataSize: '7762428B',
      failedProcessedInputDataSize: '0B',
      processedInputPositions: 60000,
      failedProcessedInputPositions: 0,
      inputBlockedTime: '72.72ms',
      failedInputBlockedTime: '0.00ns',
      outputDataSize: '1371B',
      failedOutputDataSize: '0B',
      outputPositions: 10,
      failedOutputPositions: 0,
      outputBlockedTime: '0.00ns',
      failedOutputBlockedTime: '0.00ns',
      physicalWrittenDataSize: '0B',
      failedPhysicalWrittenDataSize: '0B',
      stageGcStatistics: [Array],
      dynamicFiltersStats: [Object],
      operatorSummaries: [Array],
      optimizerRulesSummaries: [Array],
      logicalWrittenDataSize: '0B',
      writtenPositions: 0,
      spilledDataSize: '0B'
    },
    resetAuthorizationUser: false,
    setSessionProperties: {},
    resetSessionProperties: [],
    setRoles: {},
    addedPreparedStatements: {},
    deallocatedPreparedStatements: [],
    clearTransactionId: false,
    outputStage: {
      stageId: '20240711_192310_13170_mtmqs.0',
      state: 'FINISHED',
      plan: [Object],
      coordinatorOnly: false,
      types: [Array],
      stageStats: [Object],
      tasks: [Array],
      subStages: [Array],
      tables: {}
    },
    warnings: [],
    inputs: [ [Object] ],
    referencedTables: [ [Object] ],
    routines: [],
    finalQueryInfo: false,
    resourceGroupId: [ 'global', 'adhoc', 'adhoc_trino' ],
    queryType: 'SELECT',
    retryPolicy: 'QUERY',
    pruned: false,
    version: { version: '439' },
    scheduled: true,
    progressPercentage: 48.23529411764706,
    runningPercentage: 3.5294117647058822
  }
}
```